render/texture.h

```diff
@@ -1,43 +1,48 @@
 #pragma once
 
+#include <GL/glew.h>
+
 #include <string>
+#include <vector>
 
 namespace pong
 {
 
 class Texture
 {
 public:
+
+    static Texture CreateFromFile(const std::string& filePath);
+    /* Assumes the data is grayscale and stored from top left to bottom right in vector */
+    static Texture CreateFromFontCharacter(const std::vector<unsigned char>& imageData, int width, int height);
+    static Texture CreateFromSolidColor(unsigned char r, unsigned char g, unsigned char b, unsigned char a);
+
     Texture() = default;
-    explicit Texture(const std::string& filePath);
-    ~Texture();
+    Texture(const unsigned char& imageData, int width, int height, GLenum format);
+    // Copy constructors are deleted because the renderer id is used to destroy the object in opengl
+    // in the destructor. The move operator gets around this by setting the renderer id to 0.
     Texture(const Texture&) = delete;
-    Texture(Texture&&) = delete;
```

Comment:
So this is kind of interesting. Initally I had a lot of the copy/move constructors deleted for these OpenGL render classes. But I wanted to change the Texture for a Mesh to a stack variable instead, so I had to implement them all. However that caused a problem specfically when copying these objects.

Because they have a mRendererId that is created when the object on the GPU side is created. 
`glGenBuffers(1, &mRendererID);`
The destructor frees that object on the GPU side as well using that id
`glDeleteBuffers(1, &mRendererID);`

If the object is copied, the object it was copied from may destruct, and free the object from under us. So the only way for it to work without serious problems is to only implement the move constructors, and leave the copy constructors deleted. Because the move constructor will try to delete this object with an id of 0 which won't do anything
```
mRendererID = other.mRendererID;
other.mRendererID = 0;
```